mongo_connector/connector.py

```diff
@@ -59,6 +59,9 @@ def __init__(self, mongo_address, doc_managers=None, **kwargs):
         # main address - either mongos for sharded setups or a primary otherwise
         self.address = mongo_address
 
+        # connection to the main address
+        self.main_conn = None
```

Comment:
Added `main_conn` as an instance variable so the tests can mock Mongos connection failure by closing `self.main_conn`.